### ISSUE TITLE
Add template specialization edge for specialization that contains placeholder type

### ISSUE THREAD

**Issue by upsj** [CoatiSoftware/Sourcetrail#184]
When looking at the source code for a class that passes its template parameter to a base class, the displayed class name for the base class is unnecessarily complicated.
Example:

```
template<typename T>
class A {
};

template<typename T>
class B : public A<T> {
};
```

gives `A<B<typename T>::T>` as the typename of the base class, which could be simplified to `A<T>`.

This makes navigating template-heavy code difficult, since there is no direct link between `A<B<typename T>::T>` and `A<T>`

**Comment by mlangkabel** [comment]
Hi @upsj! We've put a lot of thought into this and actually it's not that simple ;)

Let me explain: There is a difference between template parameter (which is `T`) and the template argument (which is `B<typename T>::T>`). The name `A<B<typename T>::T>` indicates that whatever type has been provided to replace the template parameter of `B` is used as the template argument for the base class `A`. 

In the case you described above the solution of just calling it `T` would work, but think about cases where the template parameters of the two classes have different names or cases where the order the template parameters are defined doesn't match the order they are used as parameters. 

To handle all possible cases Coati needs to use unambiguous names. And that's the solution we came up with. If you have a better idea to tackle this problem, I'd be eager to listen ;)

**Comment by upsj** [comment]
Hey @mlangkabel! I already guessed so, since the parameter could also be of a more complicated type (think `std::vector<T>`).

Maybe I should have extended the last sentence: While `A<B<typename T>::T>` is the correct name, it's not equipped with any of the members of `A<T>` in the graph. If you added an edge from `A<B<typename T>::T>` to `A<T>` (maybe only after expanding the former node), one would probably already be able to use it just like a regular inheritance relation ;)

**Comment by mlangkabel** [comment]
We could add a template specialization edge there (because `A<B<typename T>::T>` is the instantiation of `A<T>`). 

Coati already shows those edges when a template parameter is provided with some real type as template argument (there would be an edge from `A<int>` to `A<T>`). 

For template instantiations that don't really get instantiated because they contain placeholder types it's much more complicated but we are planning on adding those edges!

**Comment by egraether** [comment]
@mlangkabel Took a quick look at this. We should add a template specialization edge from `A<B<typename T>::T>` to `A<typename T>`.

**Comment by egraether** [comment]
Fixed with template refactoring released with Sourcetrail `2019.4.61`
